extensions/eclipse-che-theia-plugin-remote/src/node/hosted-plugin-remote.ts

```diff
@@ -171,14 +183,39 @@ export class HostedPluginRemote {
       return;
     }
 
-    if (jsonMessage.method === 'getResource') {
-      const resourceBase64 = jsonMessage.data;
-      const resource = resourceBase64 ? Buffer.from(resourceBase64, 'base64') : undefined;
-      this.onGetResourceResponse(jsonMessage['pluginId'], jsonMessage['path'], resource);
+    if (InternalRequestResponsePayload.is(jsonMessage)) {
+      this.onInternalRequestResponse(jsonMessage);
       return;
     }
   }
 
+  requestPluginResourceStat(pluginId: string, resourcePath: string): Promise<Stat> {
```

Comment:
for readability maybe it would be easier to flag it as async method (and use await instead of chaining promises)

Reply:
I used to like the async/await syntax, but t.b.h these days I find it makes following the control flow rather confusing, in particular when debugging. I tend to only use async/await where it really improves readability. Here it does not, IMO: